src/test/java/com/training/assignment/tests/AssignmentTest.java

```diff
@@ -80,4 +82,33 @@ public void testOperationString() {
         softAssert.assertThat(assignment.getIndex(new String[]{"Paulo Dybala", "Federico", "Gianluigi", "Ronaldo", "Messi"}, "Ronaldo"))
                 .as("Ronaldo string not found in list").isEqualTo(3);
     }
+
+    @Test
+    public void testOperationWithLoops() {
+        softAssert.assertThat(operation.getCompanyReviews("Google")).
+                as("review not found").isEqualTo("Best platform where u will get solution for everything");
+
+        softAssert.assertThat(operation.getCompanysFeedback("Thermolab")).
+                as("Company Feedback not found").isEqualTo("good company for production of stabilizers and incubator");
+
+        softAssert.assertThat(operation.getBehaviour("good"))
+                .as("wrong decision").isEqualTo("go to heaven");
+
+        softAssert.assertThat(operation.getPlaceBehaviourBased("Bad"))
+                .as("wrong place assigned").isEqualTo("go to hell");
+
+        softAssert.assertThat(operation.getPlaceBehaviourBased("good"))
+                .as("wrong place assigned").isEqualTo("go to heaven");
+
+        softAssert.assertThat(operation.getMiddleNumber(500))
+                .as("not a middle number").isEqualTo(250);
+
```

Comment:
In .as("") method starting word, 1st letter should be alphabtetic, check for others .as()  descriptions too